Fixes broken Links in the CONTRIBUTING.MD

OberstK opened the pull request:
The links for "DRY" and the "Boy Scout Rule" were pointing to non existing sites at oreilly.com.
I replaced them with better alternatives. For the Boy Scout Rule, no better example than the new oreilly link could be found.

OberstK committed:
Fixes broken Links in the CONTRIBUTING.MD

The links for "DRY" and the "Boy Scout Rule" were pointing to non existing sites at oreilly.com.
I replaced them with better alternatives. For the Boy Scout Rule, no better example than the new oreilly link could be found.

hrhino commented on CONTRIBUTING.md:
```diff
@@ -204,8 +204,8 @@ Consider updating the package-level doc (in the package object), if appropriate.
 
-* Don't violate [DRY](http://programmer.97things.oreilly.com/wiki/index.php/Don%27t_Repeat_Yourself).
-* Follow the [Boy Scout Rule](http://programmer.97things.oreilly.com/wiki/index.php/The_Boy_Scout_Rule).
+* Don't violate [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself.
```
This line is missing the closing `)` and doesn't render properly.

hrhino commented on CONTRIBUTING.md:
```diff
@@ -204,8 +204,8 @@ Consider updating the package-level doc (in the package object), if appropriate.
-* Don't violate [DRY](http://programmer.97things.oreilly.com/wiki/index.php/Don%27t_Repeat_Yourself).
-* Follow the [Boy Scout Rule](http://programmer.97things.oreilly.com/wiki/index.php/The_Boy_Scout_Rule).
+* Don't violate [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself.
+* Follow the [Boy Scout Rule](https://www.oreilly.com/library/view/97-things-every/9780596809515/ch08.html).
```
I get a partial paywall on this site; I don't like linking to an article to explain such an important concept if that article can't all be read. 

Authoritative pages on this topic are hard to find indeed. Perhaps [this article](https://martinfowler.com/bliki/OpportunisticRefactoring.html), by the anadiplotic "Uncle Bob" Martin Fowler?

It may also be good to state the rule inline, as well.

hrhino reviewed (changes requested):
[Rendered](https://github.com/OberstK/scala/blob/patch-1/CONTRIBUTING.md).

Thanks! We have CI to prevent this on the scala-lang website, but I guess no one reads this page, so no one bothered.

OberstK commented:
I can just add an excerpt for both rules. Will switch out the O’Reilly link for the one proposed

OberstK committed:
Adding more context to the links

I also replaced the semi-paywall link with the one from Fowler (should be a reliable source and here to stay)

OberstK commented:
If the two commits are an issue, I can close the PR and reopen it with a single clean commit

SethTisue commented:
a single commit is better, yeah.

but since it's doc and not code, we don't need a merge commit, so I used GitHub's "Squash & merge" button

for future reference, you wouldn't have needed to open a new PR to get a single commit. you can squash & rebase and then force-push to the existing branch

</mechanics>

SethTisue commented:
welcome to the repo, congrats on your first scala/scala PR :-)

OberstK commented:
Thanks a lot @SethTisue! I hope I can contribute more significant changes in the future!
